packages/manager/apps/dedicated/client/app/account/user/components/newAccountForm/field/new-account-form-field-component.controller.js

```diff
@@ -0,0 +1,396 @@
+        if (this.value) {
+          this.fieldset[this.id].$setValidity('emailAvailable', isValid);
+        }
+      });
+    }
+
+    // handle special phone prefix case
+    if (this.getFieldType() === 'phone') {
+      this.$scope.$watch(
+        '$ctrl.newAccountForm.rules',
+        (rules) => {
+          const rule = rules.find(
+            (value) => value.fieldName === 'phoneCountry',
+          );
+          this.phoneCountryList = rule.in.map((country) => ({
+            country,
+            prefix: PHONE_PREFIX[country],
+            label: this.$translate.instant(`signup_enum_country_${country}`),
+          }));
+          this.phoneCountryList = this.$filter('orderBy')(
+            this.phoneCountryList,
+            'label',
+            false,
+            (a, b) => String(a.value).localeCompare(String(b.value)),
+          );
+
+          const current = this.phoneCountryList.find(
+            (value) => value.country === this.newAccountForm.model.phoneCountry,
+          );
+          const orig = this.phoneCountryList.find(
+            (value) =>
+              value.country === this.newAccountForm.originalModel.phoneCountry,
+          );
+          const country = this.phoneCountryList.find(
+            (value) => value.country === this.newAccountForm.model.country,
+          );
+          const subCountry = this.phoneCountryList.find(
+            (value) =>
+              value.country === this.newAccountForm.model.ovhSubsidiary,
+          );
+
+          this.phoneCountry =
+            current ||
+            orig ||
+            country ||
+            subCountry ||
+            head(this.phoneCountryList);
+
+          if (current !== this.phoneCountry.country) {
+            this.$timeout(() => {
+              if (this.newAccountForm.onFieldChange) {
+                this.newAccountForm.onFieldChange(
+                  {
+                    fieldName: 'phoneCountry',
+                  },
+                  this.phoneCountry.country,
+                );
+              }
+            });
+          }
+        },
+        true,
+      );
+    }
+
+    // validates input taking care of rule prefix and regularExpression
+    this.inputValidator = {
+      test: (_value) => {
+        let value = _value;
+
+        // normalize input to make it easier for the user
+        // to type numbers (ignore spaces etc)
+        if (this.isPhoneNumberField()) {
+          value = newAccountFormFieldController.cleanPhoneNumber(value);
+        } else if (this.isGenericIdentifierField()) {
+          value = newAccountFormFieldController.cleanIdentifier(value);
+        }
+
+        if (this.rule.regularExpression) {
+          if (this.rule.prefix && value) {
+            value = this.rule.prefix + value;
+          }
+          const regex = this.rule.regularExpression.replace(/\//g, '\\/');
+          return new RegExp(regex).test(value);
+        }
+        if (this.rule.fieldName === 'password') {
```

Comment:
same as https://github.com/ovh/manager/pull/8312/files#r1009210689

Reply:
It's applied